cmd/mt-kafka-mdm-report-out-of-order/inputooofinder.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/grafana/metrictank/idx/cassandra"
+	"github.com/grafana/metrictank/mdata"
+	"github.com/grafana/metrictank/mdata/errors"
+	"github.com/grafana/metrictank/schema"
+	"github.com/grafana/metrictank/schema/msg"
+	log "github.com/sirupsen/logrus"
+)
+
+type Track struct {
+	Name string
+	Tags []string
+
+	reorderBuffer *mdata.ReorderBuffer
+}
+
+type Tracker map[schema.MKey]Track
+
+// find out-of-order and duplicate metrics
+type inputOOOFinder struct {
+	prefix string
+	substr string
+
+	reorderWindow uint32
+	tracker       Tracker
+
+	groupByName             bool
+	outOfOrderGroupedByName *map[string]int
+	duplicatesGroupedByName *map[string]int
+	groupByTag              string
+	outOfOrderGroupedByTag  *map[string]int
+	duplicatesGroupedByTag  *map[string]int
```

Comment:
Seems like it might be a little cleaner if maybe `Tracker` just had a simple counter for OOO/duplicate and `inputOOOFinder` just exposes those counts. Then the business logic of grouping can be extracted from this component and done at the end when the time limit lapses.

Simplifies the critical section a little as well.